### PR TITLE
Rewrite for signals

### PR DESCRIPTION
**Pull request by mimse**
Rewrite a component a day keeps the doctor away

### CODE REVIEW

File: libs/watt/src/lib/components/breadcrumbs/watt-breadcrumbs.component.ts [hunk 1]
```diff
@@ -54,14 +51,14 @@ export class WattBreadcrumbComponent {
...
     <nav>
-      @for (breadcrumb of breadcrumbs; track breadcrumb; let isLast = $last) {
+      @for (breadcrumb of breadcrumbs(); track breadcrumb; let isLast = $last) {
         <span
           class="watt-breadcrumb"
           (click)="breadcrumb.click.emit($event)"
-          [class.interactive]="breadcrumb.click.observed"
-          [attr.role]="breadcrumb.click.observed ? 'link' : null"
+          [class.interactive]="breadcrumb.click"
+          [attr.role]="breadcrumb.click ? 'link' : null"
```
Comment (dzhavat) [inline libs/watt/src/lib/components/breadcrumbs/watt-breadcrumbs.component.ts:59]: question: doesn't this always set `link` `role`/`interactive` class on the element?

Comment (mimse) [reply libs/watt/src/lib/components/breadcrumbs/watt-breadcrumbs.component.ts:59]: Think you are right, buuuut https://github.com/angular/angular/issues/54837

Comment (mimse) [reply libs/watt/src/lib/components/breadcrumbs/watt-breadcrumbs.component.ts]: It should be fixed now :-)

File: libs/watt/src/lib/components/breadcrumbs/watt-breadcrumbs.component.ts [hunk 2]
```diff
@@ -36,10 +35,10 @@ import { WattIconComponent } from '../../foundations/icon/icon.component';
   template: `<ng-template #templateRef><ng-content /></ng-template>`,
 })
 export class WattBreadcrumbComponent {
-  @ViewChild('templateRef', { static: true }) public templateRef!: TemplateRef<unknown>;
-
+  templateRef = viewChild.required<TemplateRef<unknown>>('templateRef');
   // Used to determine if the breadcrumb is interactive or not
-  @Output() click: EventEmitter<unknown> = new EventEmitter<unknown>(); // eslint-disable-line @angular-eslint/no-output-native
+  actionEmitter = new Subject<unknown>();
+  click = outputFromObservable(this.actionEmitter);
```
Comment (dzhavat) [inline libs/watt/src/lib/components/breadcrumbs/watt-breadcrumbs.component.ts:41]: praise: nice workaround until we get a dedicated property

**Review by dzhavat — APPROVED** [review]

**Pull request merged**